File: app/src/main/java/vn/loitp/app/activity/customviews/layout/elasticdragdismisslayout/ElasticDragDismissLayoutActivity.java
package vn.loitp.app.activity.customviews.layout.elasticdragdismisslayout;

import android.os.Bundle;
import android.widget.TextView;

import com.core.base.BaseFontActivity;

import loitp.basemaster.R;
import vn.loitp.views.layout.elasticdragdismisslayout.ElasticDragDismissCallback;
import vn.loitp.views.layout.elasticdragdismisslayout.ElasticDragDismissLinearLayout;

//https://github.com/Commit451/ElasticDragDismissLayout?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=3098
public class ElasticDragDismissLayoutActivity extends BaseFontActivity {
    private TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        tv = (TextView) findViewById(R.id.tv);
        ElasticDragDismissLinearLayout elasticDragDismissLinearLayout = (ElasticDragDismissLinearLayout) findViewById(R.id.draggable_frame);
        elasticDragDismissLinearLayout.addListener(new ElasticDragDismissCallback() {
            @Override
            public void onDrag(float elasticOffset, float elasticOffsetPixels, float rawOffset, float rawOffsetPixels) {
                tv.setText("onDrag " + elasticOffset + " - " + elasticOffsetPixels + " - " + rawOffset + " - " + rawOffsetPixels);
            }

            @Override
            public void onDragDismissed() {
                tv.setText("onDragDismissed");

                //if you are targeting 21+ you might want to finish after transition
                finish();
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_elasticdragdismisslayout;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/layout/elasticdragdismisslayout/ElasticDragDismissLayoutActivity.kt
package vn.loitp.app.activity.customviews.layout.elasticdragdismisslayout

import android.annotation.SuppressLint
import android.os.Bundle
import com.annotation.IsFullScreen
import com.annotation.LayoutId
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.views.layout.elasticdragdismisslayout.ElasticDragDismissCallback
import kotlinx.android.synthetic.main.activity_elasticdragdismisslayout.*
import vn.loitp.app.R

//https://github.com/Commit451/ElasticDragDismissLayout?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=3098
@LayoutId(R.layout.activity_elasticdragdismisslayout)
@LogTag("ElasticDragDismissLayoutActivity")
@IsFullScreen(false)
class ElasticDragDismissLayoutActivity : BaseFontActivity() {

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        draggable_frame.addListener(object : ElasticDragDismissCallback() {

            override fun onDrag(elasticOffset: Float, elasticOffsetPixels: Float, rawOffset: Float, rawOffsetPixels: Float) {
                textView.text = "onDrag $elasticOffset - $elasticOffsetPixels - $rawOffset - $rawOffsetPixels"
            }

            override fun onDragDismissed() {
                textView.text = "onDragDismissed"

                //if you are targeting 21+ you might want to finish after transition
                finish()
            }
        })
    }

}
